src/aspire/utils/misc.py

```diff
@@ -318,7 +318,10 @@ def pairs_to_linear(n, i, j):
     Converts from all_pairs indexing (i, j), where i<j, to linear indexing.
     ie. (0, 1) --> 0 and (n-2, n-1) --> n * (n - 1)/2 - 1
     """
-    assert i < j < n, "i must be less than j, and both must be less than n."
+    i = np.array(i)
+    j = np.array(j)
+
+    assert (i < j).all() < n, "i must be less than j, and both must be less than n."
```

Comment:
are you sure this line does what you think?

Reply:
This is __not__ doing what I thought it was. Thanks for pointing that out!

Yeah, maybe I should use the dict. We are calling `pairs_to_linear` no less than `3 * n-choose-3` times, so alot.

Reply:
hrrm. smells like a bake off.  just in time for friday.

i) formula in python

ii) a dict (i,j) -> n.

iii) n,n array mapping i,j -> n.

This is one of those times where in a low level language (think C) this formula might be preferable, but that is a lot of goo happening in python to get the index.

Reply:
the index mapping can probably be uint16 to save space 😇 .

Reply:
Array mapping for the win ⚡

Reply:
It bought 4 or 5 seconds

Reply:
This will really pay off for larger runs of the algorithm though